cli/lsp/language_server.rs

```diff
@@ -802,6 +802,48 @@ impl lspower::LanguageServer for LanguageServer {
     }
   }
 
+  async fn rename(
+    &self,
+    params: RenameParams,
+  ) -> LSPResult<Option<WorkspaceEdit>> {
+    let specifier =
+      utils::normalize_url(params.text_document_position.text_document.uri);
+    let line_index = match self.get_line_index(specifier.clone()).await {
+      Ok(line_index) => line_index,
+      Err(_) => return Err(LSPError::internal_error()),
+    };
+
+    let req = tsc::RequestMethod::FindRenameLocations((
+      specifier,
+      text::to_char_pos(&line_index, params.text_document_position.position),
+      true,
+      true,
+      false,
+    ));
+
+    let res = match self.ts_server.request(self.snapshot(), req).await {
+      Ok(res) => res,
+      Err(_) => return Err(LSPError::invalid_request()),
```

Comment:
Instead of swallowing errors, could you log them with `error!` here? Also this might be a bit cleaner if you used `.map_err` instead of `match`.

Same goes for the other places you return errors.

Reply:
I'm a rust newbie, so I'll confirm with a code example.
Is the code below ok?

```rust
    let res = self.ts_server.request(self.snapshot(), req).await.map_err(|err| {
      error!("Failed to request to tsserver {}", err);
      return LSPError::invalid_request();
    })?;
```

Reply:
In Rust, the last statement of a block without a semicolon is an implied return value, so you should do:

```rust
    let res = self.ts_server.request(self.snapshot(), req).await.map_err(|err| {
      error!("Failed to request to tsserver {}", err);
      LSPError::invalid_request()
    })?;
```

Reply:
Thank you! I've fixed it.